forbidigo/forbidigo.go

```diff
@@ -97,12 +106,29 @@ type visitor struct {
 	linter   *Linter
 	comments []*ast.CommentGroup
 
-	fset   *token.FileSet
-	issues []Issue
+	fset      *token.FileSet
+	typesInfo *types.Info
+	issues    []Issue
 }
 
-func (l *Linter) Run(fset *token.FileSet, nodes ...ast.Node) ([]Issue, error) {
-	var issues []Issue 
+func (l *Linter) RunForPackages(pkgs []*packages.Package) ([]Issue, error) {
```

Comment:
Am I right in thinking we've put this here to avoid having to repeat this in our test code?  In the process, we are also bringing in `"golang.org/x/tools/go/packages"` as a dependency.  My preference would just be to duplicate the code we have to in the tests to keep the extra dependency out.  I think it's less about the extra dependency and just about keeping this package about doing the minimal thing that needs to happen to analyze the code so that it could be plugged in anywhere.  Every linter presumably has this same need (to iterate over packages).

Reply:
Besides avoiding code duplication my reasoning also was that now the unit test covers more of the tool's logic by using the same code as the binary. But I don't feel strongly about this. I can also duplicate the code.

Avoiding the dependency is harder. Then I would have to find some other way of populating the additional structs.

Perhaps it is time to accept that unit test coverage for this particular package lives elsewhere (at the root for stand-alone usage and in pkg/analyzer)? Then the forbidigo.go file here can drop the dependency and ../main_test.go (aka the current forbidigo_test.go) can still use it.

Reply:
Or leave `golang.org/x/tools/go/packages` out of the non-test code and only use it in the `logr_test.go`? Then downstream users of the package are not affected by the additional dependency.

Just let me know what you prefer and I'll change it.

Reply:
I have no objections to including it in the test code.

Reply:
Then let's do that - pushed.